index.js

```diff
@@ -75,4 +75,28 @@ exports.hash = function(pwd, salt, fn){
   return function(done){
     fn = done;
   }
-};
\ No newline at end of file
+  
+/**
+ * Hashes asynchronously a password with optional `salt`, otherwise
+ * generate a salt for `pass` and invoke `fn(err, salt, hash)`.
+ *
+ * @param {String} password to hash
+ * @param {String} optional salt
+ * @param {Function} callback
+ * @api public
+ */
+exports.hashAsync = function(pwd, salt, fn){
```

Comment:
needs a new name as well, both are async. Might be nicer to break compatibility with the current "thunk" version (https://github.com/iquirino/node-pwd/blob/patch-1/index.js#L75-L77) which allows `yield pwd.hash(p, s)` with the function returned, by returning a promise there instead